### PR TITLE
add notifications

### PR DESCRIPTION
**Pull request by Ozzorany**
image

image

image

image

image

### CODE REVIEW

File: src/app/barter/effects/transactions.effects.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ export class TransactionsEffects {
     )
   );
 
+  observeTransactions$ = createEffect(() => this.transactionsService.getAll$().pipe(skip(1),
```
Comment (yarivg) [inline src/app/barter/effects/transactions.effects.ts:35]: why skip?

File: src/app/barter/services/transactions.service.ts [hunk 2]
```diff
@@ -29,4 +32,24 @@ export class TransactionsService implements IFirebaseService<Transaction> {
...
   }
+
+  getNewTransactions$(): Observable<Transaction[]> {
+    return this.newTransaction$.asObservable();
+  }
+
+  alertNewTransaction(transactions: Transaction[]) {
+    this.userStore$.select(getUser).subscribe((user: User) => {
+      const myNewTransactions: Transaction[] = transactions.filter(transaction =>
+        this.shouldFilterTransaction(!!transaction.transactionCompleteDate ? transaction.transactionCompleteDate.toDate() :
```
Comment (yarivg) [inline src/app/barter/services/transactions.service.ts:43]: Does it also filter notification by user? 
Example of what shouldnt happen - my user gets a notification - for a barter offer of shperling to you

File: src/app/components/navbar/navbar.component.ts [hunk 3]
```diff
@@ -15,13 +20,56 @@ import {logout} from '../../user/actions/user.actions';
...
+    this.transactionService.getNewTransactions$().pipe(mergeMap((transactions: Transaction[]) =>
+      forkJoin(
+        transactions.map((transaction: Transaction) => forkJoin({
+          transactionId: of(transaction.id),
+          description: of(this.getDescription(transaction.status)),
+          trader: this.userService.getById$(transaction.operatedBy).pipe(first()),
+          date: of(!!transaction.transactionCompleteDate ? transaction.transactionCompleteDate.toDate() : transaction.offerDate.toDate())
+        })))
+    )).subscribe((alerts: Alert[]) => {
+      this.alerts = alerts;
```
Comment (yarivg) [inline src/app/components/navbar/navbar.component.ts]: can make it one line without assignment 
this.alerts = alert.sort((a: Alert, b: Alert) => b.date.getTime() - a.date.getTime());

**Review by yarivg — APPROVED** [review]
Look at what I commented..

File: src/app/barter/effects/transactions.effects.ts (hunk 1 above)
Comment (Ozzorany) [reply src/app/barter/effects/transactions.effects.ts:35]: For skipping the first time it's happening, when the app is up for the first time

File: src/app/barter/services/transactions.service.ts (hunk 2 above)
Comment (Ozzorany) [reply src/app/barter/services/transactions.service.ts:43]: done

File: src/app/components/navbar/navbar.component.ts (hunk 3 above)
Comment (Ozzorany) [reply src/app/components/navbar/navbar.component.ts]: done

**Pull request merged**